pandas/core/arrays/numpy_.py

```diff
@@ -229,13 +229,15 @@ def __getitem__(self, item):
     def __setitem__(self, key, value):
         value = extract_array(value, extract_numpy=True)
 
-        if not lib.is_scalar(key) and is_list_like(key):
+        scalar_key = lib.is_scalar(key)
+        scalar_value = lib.is_scalar(value)
+
+        if not scalar_key and is_list_like(key):
             key = np.asarray(key)
 
-        if not lib.is_scalar(value):
-            value = np.asarray(value)
+        if not scalar_value:
```

Comment:
it looks like this is only used once, can re-inline?  Also looks like `not scalar_key` is redundant with `is_list_like(key)`.  If the latter is not the case, I'd be curious why that is

Reply:
Resolved, I think. I was passing the wrong thing to `is_list_like`.

>  If the latter is not the case, I'd be curious why that is

It's irrelevant to this PR, but to satiate your curiosity, the difference is 0-dim ndarrays.  They are apparently neither list-like nor scalars.

```python
In [17]: a = np.array(1)

In [18]: pd._libs.lib.is_scalar(a)
Out[18]: False

In [19]: pd.api.types.is_list_like(a)
Out[19]: False
```